ctl/vr/inductive_complex.h

```diff
@@ -0,0 +1,130 @@
+* NOTES
+* This is the construction of the Vietoris Rips Complex using the all pairs
+* approach in the geometric phase and the inductive algorithm in the
+* topological phase.
+*******************************************************************************/
+
+//STL
+#include <set>
+#include <vector> 
+#include <algorithm>
+
+//BOOST
+#include <boost/graph/adjacency_list.hpp>
+
+//CTL
+#include <ctl/vr/incremental_complex.h>
+
+namespace ctl {
+
+// We add the given graph to the complex. Also, we update the cells vectors to
+// keep track of the dimensions of the cells.
+template<typename Graph, typename Complex, typename Vector>
+void add_graph_to_complex (const Graph& g,
+			   Complex& complex,
+			   Vector& k_cells,
+			   Vector& k_plus_one_cells) {
+    typedef typename Complex::Cell Simplex;
+    typedef typename boost::graph_traits< Graph>::vertex_iterator vertex_iterator;
+    typedef typename boost::graph_traits< Graph>::adjacency_iterator Iterator;
+    vertex_iterator vi, vlast;
+    for ( std::tie( vi, vlast) = boost::vertices( g); vi != vlast; ++vi) {
+        // adding {*vi} to both k_cells and complex
+        Simplex tau(1, *vi);
+        k_cells.push_back(tau);
+        complex.insert_open_cell(tau);
+        Iterator ai, ai_end;
+        for ( boost::tie(ai, ai_end) = boost::adjacent_vertices(*vi, g);
+              ai != ai_end; ++ai) {
+                // adding {*vi, *ai} to both k_plus_one_cells and complex
+                Simplex sigma( tau);
+                sigma.insert( *ai);
+                k_plus_one_cells.push_back(sigma);
+                complex.insert_open_cell(sigma);
+        }
+    }
+}
+
+// We loop through all nodes, selecting each one as the starting point for
+// constructing our simplicial complex.
+template<typename Graph, typename Complex> 
+void inductive_vr (const Graph& g, Complex& complex, std::size_t dimension) { 
+    typedef typename Complex::Cell Simplex;
+    typedef typename Simplex::value_type Vertex;
+    std::vector<Simplex> k_cells;
+    std::vector<Simplex> k_plus_one_cells;
+    // adding the graph to the complex
+    add_graph_to_complex (g, complex, k_cells, k_plus_one_cells);
+    // inductive algorithm
+    for ( int k = 1; k < dimension; k++) {
+        k_cells.swap(k_plus_one_cells);
+	k_plus_one_cells.clear();
+	for( const auto & tau: k_cells){
+	    // getting the intersection of all lower neighbors
+	    std::vector< Vertex> final_neighbors;
+            std::vector< Vertex> lower_neighbors;
+	    ctl::get_lower_neighbors(g, *(tau.begin()), final_neighbors);
+	    for( auto & vi: tau) {
+		if( final_neighbors.empty()) { break; }
+		lower_neighbors.clear();
+		ctl::get_lower_neighbors( g, vi, lower_neighbors);
+		std::set< Vertex> intersect;
+		set_intersection(lower_neighbors.begin(),
+				 lower_neighbors.end(),
+				 final_neighbors.begin(),
+				 final_neighbors.end(),
+				 std::inserter(intersect,intersect.begin()));
```

Comment:
Why did you change back inserter? Never insert into the front of a vector! Do you understand why?